### PR TITLE
Added permissions

### PR DESCRIPTION
**Pull request by gourneau**
Added basic permissions (array of strings). Because sometimes you might want to disable multiple things

It could be used like this 

``` javascript
            $.hulk("#settings",  window.settings, function(data){
                console.log("Here's the updated data: " + data);
            }, {
                "showSaveButton" : false,
                "emptyString" : true,
                "permissions" : ["hide-toggle","no-append"]
            });
```

### CODE REVIEW

File: hulk.js [hunk 1]
```diff
@@ -156,20 +165,27 @@
...
         valueHTML.addClass('hulk-map-value-container');
         if (valueHTML.children('.hulk-map-pair, .hulk-array-element').length > 0) {
             var button = $(document.createElement('button'));
             button.addClass('hulk-collapse-item');
             if (depth !== 0) {
                 button.text("Collapse");
+
                 attachCollapseHandler(button);
-                pair.append(button);
+                if ($.inArray("hide-toggle", hideToggle) === -1) {
```
Comment (kevinburke) [inline hulk.js]: Wondering if we should move this logic higher up... do we even want to create the button if `hide-toggle` is off?

**Comment by kevinburke** [comment]
This looks good! I haven't used this project in a while sadly... assuming it works but did you do basic checking ?

**Comment by gourneau** [comment]
Yup, I am using it for an internal tool.

**Comment by kevinburke** [comment]
Gah, I'm really sorry... I should just merge this

**Pull request closed without merging**